files/external_node_v2.rb

```diff
@@ -151,7 +151,7 @@ def generate_fact_request(certname, filename)
   # Temp file keeping the last run time
   stat = stat_file("#{certname}-push-facts")
   last_run = File.exists?(stat) ? File.stat(stat).mtime.utc : Time.now - 365*24*60*60
-  last_fact = File.stat(filename).mtime.utc
+  last_fact = File.exists?(filename) ? File.stat(filename).mtime.utc : Time.new(0)
```

Comment:
Ah! Thanks for pointing that out; i'll fix it later today
________________________________


Time.new doesn't take any arguments in Ruby 1.8.7<https://ruby-doc.org/core-1.8.7/Time.html#method-c-new>, it might be best to use Time.at(0) (returns Unix epoch, 1970).